server/hedley/modules/custom/hedley_reports/hedley_reports.module

```diff
@@ -5674,193 +5680,3072 @@ function hedley_reports_generate_completion_data_for_tuberculosis($participant,
 }
 
 /**
- * Checks if a lab test was performed based on the test execution note.
- *
- * Determines if the test execution note indicates that the test was performed
- * by checking if it matches one of the predefined statuses.
- *
- * @param object $test
- *   An object representing the lab test, which contains the test execution
- *   note field.
+ * Generates completion data for acute prenatal encounters during pregnancy.
  *
- * @return bool
- *   TRUE if the test was performed based on the execution note, FALSE
- *   otherwise.
+ * @param object $participant
+ *   The participant node object representing pregnancy.
+ * @param bool $exclude_set
+ *   Indicate whether to exclude encounters with report data already set.
  */
-function hedley_reports_lab_test_performed_by_execution_note($test) {
-  if (empty($test) || empty($test->field_test_execution_note)) {
-    return FALSE;
-  }
+function hedley_reports_generate_completion_data_for_prenatal($participant, $exclude_set) {
```

Comment:
perhaps this is a function that we can simplify and make it shorter, like extracting the mapping out of it, etc.

Reply:
Well, mapping is a simple component that can collapsed in the editor, so I don't see extra value there, but I did take out some other parts into separate functions.